app/Filament/Resources/TimeScheduleResource.php

```diff
@@ -0,0 +1,153 @@
+use App\Models\TimeSchedule;
+use App\Models\User;
+use Carbon\Carbon;
+use Filament\Forms;
+use Filament\Forms\Components\Section;
+use Filament\Forms\Components\Select;
+use Filament\Forms\Components\TimePicker;
+use Filament\Forms\Form;
+use Filament\Forms\Get;
+use Filament\Resources\Resource;
+use Filament\Tables;
+use Filament\Tables\Columns\TextColumn;
+use Filament\Tables\Table;
+use Illuminate\Database\Eloquent\SoftDeletingScope;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+
+class TimeScheduleResource extends Resource
+{
+    protected static ?string $model = TimeSchedule::class;
+
+    protected static ?string $navigationIcon = 'heroicon-o-rectangle-stack';
+
+    public static function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                Section::make('')
+                    ->schema([
+                        Select::make('role_id')
+                            ->label('Schedule For')
+                            ->options(DB::table('roles')->where('name', '!=', User::ROLE_ADMIN)->pluck('name', 'id'))
+                            ->required()
+                            ->native(false)
+                            ->preload()
+                            ->live()
+                            ->searchable(),
+                        Select::make('user_id')
+                            ->label('User')
+                            ->options(fn (Get $get): Collection => User::query()->withWhereHas('roles', function ($query) use ($get) {
+                                return $query->where('id', $get('role_id'));
+                            })->pluck('name', 'id'))
+                            ->required()
+                            ->native(false)
+                            ->searchable()
+                    ])->columns()->visible(fn ($record) => $record === null),
+                Section::make('')
+                    ->schema([
+                        TimePicker::make('start_time')
+                            ->datalist([
+                                '07:00',
+                                '08:00',
+                                '09:00',
+                                '15:00',
+                                '16:00',
+                                '22:00',
+                                '00:00'
+                            ])
+                        ->seconds(false),
+                        TimePicker::make('end_time')
+                            ->datalist([
+                                '15:00',
+                                '16:00',
+                                '17:00',
+                                '18:00',
+                                '19:00',
+                                '20:00',
+                                '22:00',
+                            ])
+                            ->seconds(false),
+                        Select::make('day_off_number')
```

Comment:
@kolanaveen As this field is not nullable in migration then we need to add `required` validation here.